### PR TITLE
[WIP] Re-synchronizing data loaders

### PR DESCRIPTION
**Pull request by alecgunny**
Hacking around the `ChunkBuffer` class to return synchronous functionality to data loaders in light of performance regressions from asynchronous loading. Ideally, some of this functionality can be retained long-term to make asychronous data loading an `__init__` kwarg on the `DataLoader` class, in case users don't want it for some reason

### CODE REVIEW

File: nvtabular/loader/backend.py [hunk 1]
```diff
@@ -142,9 +177,13 @@ def load_chunks(self, dev, dataloader):
                         spill = workflow.apply_ops(spill)
                     spill = dataloader._create_tensors(spill)
                     spill = dataloader._handle_tensors(*spill)
-                    self.put([spill])
+                    # ADDING FOR SYNC BEHAVIOR
+                    # self.put([spill])
+                    yield spill
```
Comment (benfred) [inline nvtabular/loader/backend.py:182]: Should this be ```yield [spill]``` instead to match previous behaviour?

File: nvtabular/loader/backend.py [hunk 2]
```diff
@@ -55,16 +74,28 @@ def __init__(
         self.q_out = queue.Queue(qsize)
         self._stop_event = threading.Event()
 
+        # adding for sync behavior
+        self._itr = None
+        self._iterating = False
```
Comment (benfred) [inline nvtabular/loader/backend.py:79]: Can we just use ```self._itr is None``` as the check to see if we're iterating ?

File: nvtabular/loader/backend.py (hunk 1 above)
Comment (alecgunny) [reply nvtabular/loader/backend.py:182]: ah good catch yes it should

File: nvtabular/loader/backend.py (hunk 2 above)
Comment (alecgunny) [reply nvtabular/loader/backend.py:79]: Yeah that actually might make more sense I'll give that a test

**Comment by jperez999** [comment]
rerun tests

**Comment by alecgunny** [comment]
@jperez999 do these torch test failures help elucidate what might be going wrong at all?

**Review by jperez999 — APPROVED** [review]

**Pull request merged**